rllib/policy/eager_tf_policy.py

```diff
@@ -136,16 +123,21 @@ def compute_actions_from_input_dict(
                 episodes: Optional[List[Episode]] = None,
                 **kwargs
         ) -> Tuple[TensorType, List[TensorType], Dict[str, TensorType]]:
+            """Traced version of Policy.compute_actions_from_input_dict."""
 
-            # Create a traced version of `self._compute_action_helper`.
-            if self._traced_compute_action_helper is False:
-                self._compute_action_helper = convert_eager_inputs(
+            self._check_too_many_retraces()
+
+            # Create a traced version of `self._compute_actions_helper`.
+            if self._traced_compute_actions_helper is False:
+                self._compute_actions_helper = convert_eager_inputs(
```

Comment:
question, why don't we simply init self._traced_compute_actions_helper to None, and let self._traced_compute_actions_helper be the traced function that gets used all the time?

Reply:
I wanted to utilize the super()'s (eager_tf_policy's) method's logic (which doesn't call `_traced_compute_actions_helper` b/c it doesn't know about it), so we don't have to copy all the code.

So we basically
1) trace the helper method and assign it back to "itself", such that the traced eager policy has `self._compute_actions_helper` as a traced function, but non-traced eager policy still has it as untraced function.
2) Now just call the super()'s actual method.

This way, we can share all the "preprocessing" logic, like zero-padding, generating the tensors, etc.., and only then calling the traced (or untraced) helper method.
This allowed us to disentangle all this messy: ..._eager, ..._traced_eager, `_compute_actions`, etc.. It was impossible to understand which method gets called by which other one. It's still complex, but much simpler now.